examples/demo-nft-module/src/call.rs

```diff
@@ -2,94 +2,314 @@ use anyhow::{bail, Result};
+                bail!(
+                    "Collection with name {} by sender {} is frozen and cannot be updated",
+                    collection_name,
+                    creator.to_string()
+                );
+            } else {
+                c.collection_uri = collection_uri.to_string();
+                self.collections.set(&collection_address, &c, working_set);
+            }
+        } else {
+            bail!(
+                "Collection with name {} by sender {} does not exist",
+                collection_name,
+                creator.to_string()
+            );
+        }
+        Ok(CallResponse::default())
+    }
+
+    pub(crate) fn freeze_collection(
+        &self,
+        collection_name: &str,
+        context: &C,
+        working_set: &mut WorkingSet<C::Storage>,
+    ) -> Result<CallResponse> {
+        let creator = context.sender();
+        let collection_address = get_collection_address::<C>(collection_name, creator.as_ref());
 
-        working_set.add_event("NFT mint", &format!("A token with id {id} was minted"));
+        if let Some(mut c) = self.collections.get(&collection_address, working_set) {
+            if c.frozen {
+                bail!(
+                    "Collection with name {} by sender {} is already frozen",
+                    collection_name,
+                    creator.to_string()
+                )
+            } else {
+                c.frozen = true;
+                self.collections.set(&collection_address, &c, working_set);
+            }
+        } else {
+            bail!(
+                "Collection with name {} by sender {} does not exist",
+                collection_name,
+                creator.to_string()
+            );
+        }
         Ok(CallResponse::default())
     }
 
-    pub(crate) fn transfer(
+    #[allow(clippy::too_many_arguments)]
+    pub(crate) fn mint_nft(
         &self,
-        id: u64,
-        to: C::Address,
+        token_id: u64,
+        collection_name: &str,
+        collection_uri: &str,
+        mint_to_address: &UserAddress<C>,
+        frozen: bool,
         context: &C,
         working_set: &mut WorkingSet<C::Storage>,
     ) -> Result<CallResponse> {
-        let token_owner = match self.owners.get(&id, working_set) {
-            None => {
-                bail!("Token with id {} does not exist", id);
+        let creator = context.sender();
+        let collection_address = get_collection_address::<C>(collection_name, creator.as_ref());
+
+        if let Some(mut c) = self.collections.get(&collection_address, working_set) {
+            let nft_identifier = NftIdentifier(token_id, collection_address.clone());
+            if c.frozen {
+                bail!(
+                    "Collection with name {} by sender {} is already frozen",
+                    collection_name,
+                    creator.to_string()
+                )
+            } else if self.nfts.get(&nft_identifier, working_set).is_some() {
```

Comment:
Nit I think we could avoid `3-branch if` in the following way:
1. The first ` if c.frozen` just early returns so we don'y need the second else_if
2. For the second and third else we use explicit pattern matching on Option.